clarity/src/vm/functions/assets.rs

```diff
@@ -840,7 +840,7 @@ pub fn special_burn_token(
 
     if let (Value::UInt(amount), Value::Principal(ref burner)) = (amount, from) {
         if amount == 0 {
-            return clarity_ecode!(MintTokenErrorCodes::NON_POSITIVE_AMOUNT);
+            return clarity_ecode!(BurnTokenErrorCodes::NON_POSITIVE_AMOUNT);
```

Comment:
I think this would be a consensus-breaking change: `MintTokenErrorCodes::NON_POSITIVE_AMOUNT` is `1`, while `BurnTokenErrorCodes::NON_POSITIVE_AMOUNT` is `3`.

I think the right thing to do here is to redefine `BurnTokenErrorCodes` as:

```rust
enum BurnTokenErrorCodes {
    NOT_ENOUGH_BALANCE_OR_NON_POSITIVE = 1,
}
```

And use that variant in both places for the `BurnTokenErrorCodes`.